perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3762,10 +3765,60 @@ public void updateExistingMemberWhileSynchronization(PerunSession sess, Group gr
 		//Synchronize userExtSources (add not existing)
 		addUserExtSources(sess, candidate, memberToUpdate);
 
+		if (isGroupMember(sess, group, memberToUpdate)) {
```

Comment:
Be careful. This method checks whether the member is in the group, but it does not care if the membership type is direct or indirect. I think we care only about direct memberships.